### ISSUE TITLE
Question: [@typescript-eslint/naming-convention] Disable rule for certain typescript types

### ISSUE THREAD

**Issue by Nkmol** [typescript-eslint/typescript-eslint#5698]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.3&sourceType=module&showAST=es&code=MYewdgzgLgBAZgUwQEwPICMBWDhQFwwAeAtgDYB0AGmRtrjALwwDeAUDDAE4QQEDa7DixgB9AIZQonAsxgA3BNwCW4AgHIATOQAMagDQw1JUpDwSQxNeoAWkgA54A9I4Dub8i4DM5EJwDmjhra2gCsjgCCUBZqMAC+cXqCALqJsQDcQA

### Repro Code

```typescript
const feedObject = {
   rss: [
      { _attr: { version: '2.0', 'xmlns:atom': 'http://www.w3.org/2005/Atom' } },
   ],
};
```

### ESLint Config

```json
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "warnOnUnsupportedTypeScriptVersion": false,
    "project": "tsconfig.json",
    "sourceType": "module",
    "ecmaVersion": "latest"
  },
  "rules": {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "TSQualifiedName[name=XmlObject]",
        "format": null
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "module": "node16",
  "compilerOptions": {
    "moduleResolution": "node16",
    "resolveJsonModule": true,
    "types": [
      "node"
    ],
  },
}
```

### Expected Result

I would expect the "naming-convention" rule to be disabled on any object of type `XmlObject`.

### Actual Result

> BaseConfig: Configuration for rule "@typescript-eslint/naming-convention" is invalid: Value "TSQualifiedName[name=XmlObject]" should be array. Value "TSQualifiedName[name=XmlObject]" should be equal to one of the allowed values.

### Additional Info

Am I safe to assume the `@typescript-reslint/*` rule do not allow for the full AST identifier selectors? I foud the following example, which is a ESLint rule, which does seem to correctly parse: https://typescript-eslint.io/docs/linting/troubleshooting/#how-can-i-ban-specific-language-feature

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.35.1` |
| `@typescript-eslint/parser`            | `5.35.1` |
| `TypeScript`                           | `4.8.3` |
| `ESLint`                               | `8.22.0` |
| `node`                                 | `16.17.0` |
| `xo`                                 | `0.52.3` |

**Comment by bradzacher** [comment]
It does not allow for arbitrary AST selectors.
The only supported selectors are listed in the docs.

The docs also list ways in which you can allowlist/ignore specific names. Check out the examples under this heading
https://typescript-eslint.io/rules/naming-convention/#ignore-properties-that-require-quotes

---

Arbitrary AST selectors wouldn't make sense in the context of the rule IMO - the basic cases are already covered by the selectors and modifiers in the rule, and the advanced cases would be too hyper-specific to be useful beyond one or two cases.
